src/DGtal/images/ImageContainerBySTLMap.h

```diff
@@ -78,102 +79,109 @@ namespace DGtal
+   * - through the range of values returned by the range() method,
    * which can be used to directly iterate over the values of the image
    *
-   * This class also provides a setValue() method and an output iterator, 
-   * which is returned by the outputIterator() method for writting purposes. 
+   * This class also provides a setValue() method and an output iterator,
+   * which is returned by the outputIterator() method for writting purposes.
    *
    * @see testImage.cpp
    */
 
   template <typename TDomain, typename TValue>
-  class ImageContainerBySTLMap: 
+  class ImageContainerBySTLMap:
     public std::map<typename TDomain::Point, TValue >
   {
 
   public:
 
-    typedef ImageContainerBySTLMap<TDomain,TValue> Self; 
-    typedef std::map<typename TDomain::Point, TValue > Parent; 
+    typedef ImageContainerBySTLMap<TDomain,TValue> Self;
+    typedef std::map<typename TDomain::Point, TValue > Parent;
 
     /// domain
     BOOST_CONCEPT_ASSERT(( concepts::CDomain<TDomain> ));
-    typedef TDomain Domain;    
+    typedef TDomain Domain;
     typedef typename Domain::Point Point;
     typedef typename Domain::Vector Vector;
     typedef typename Domain::Integer Integer;
     typedef typename Domain::Size Size;
     typedef typename Domain::Dimension Dimension;
     typedef Point Vertex;
 
-    // Pointer to the (const) Domain given at construction.
-    typedef const Domain * DomainPtr;
-
-
     /// static constants
     static const typename Domain::Dimension dimension;
 
     /// range of values
     BOOST_CONCEPT_ASSERT(( CLabel<TValue> ));
     typedef TValue Value;
-    typedef DefaultConstImageRange<Self> ConstRange; 
-    typedef DefaultImageRange<Self> Range; 
+    typedef DefaultConstImageRange<Self> ConstRange;
+    typedef DefaultImageRange<Self> Range;
 
     /// output iterator
-    typedef SetValueIterator<Self> OutputIterator; 
+    typedef SetValueIterator<Self> OutputIterator;
 
     /////////////////// Data members //////////////////
-  private: 
+  private:
 
-    /// Counted pointer on the image domain,
+    /// Shared pointer on the image domain,
     /// Since the domain is not mutable, not assignable,
     /// it is shared by all the copies of *this
-    DomainPtr myDomainPtr;
+    std::shared_ptr<const Domain> myDomainPtr;
 
     /// Default value
     Value myDefaultValue;
 
     /////////////////// standard services //////////////////
 
-  public: 
+  public:
 
-    /** 
-     * Constructor from a Domain
-     * 
+    /**
+     * Constructor from a Domain.
+     *
+     * If Domain is a heavy type, prefer giving a smart pointer on the domain.
+     *
      * @param aDomain the image domain.
      * @param aValue a default value associated to the domain points
      * that are not contained in the underlying map.
      */
-    ImageContainerBySTLMap( ConstAlias<Domain> aDomain, const Value& aValue = 0);
+    ImageContainerBySTLMap( const Domain & aDomain, const Value& aValue = 0);
```

Comment:
Using here `ConstAlias` would disallow inline construction of the domain (it is sometimes the case in DGtal). I could also pass it by value but since domains are not moveable, this will imply an additional copy...

I can also make it more clear in the documentation that the domain is copied and not aliased.